src/roles/dullahan.py

```diff
@@ -5,76 +5,73 @@
 
 import src.settings as var
 from src.utilities import *
-from src import debuglog, errlog, plog
-from src.decorators import cmd, event_listener
+from src.functions import get_players, get_target
+from src import users, debuglog, errlog, plog
+from src.decorators import command, event_listener
 from src.messages import messages
 from src.events import Event
 import botconfig
 
-KILLS = {} # type: Dict[str, str]
-TARGETS = {} # type: Dict[str, Set[str]]
+KILLS = {} # type: Dict[users.User, users.User]
+TARGETS = {} # type: Dict[users.User, Set[users.User]]
 
-@cmd("kill", chan=False, pm=True, playing=True, silenced=True, phases=("night",), roles=("dullahan",))
-def dullahan_kill(cli, nick, chan, rest):
+@command("kill", chan=False, pm=True, playing=True, silenced=True, phases=("night",), roles=("dullahan",))
+def dullahan_kill(var, wrapper, message):
     """Kill someone at night as a dullahan until everyone on your list is dead."""
-    if not TARGETS[nick] & set(list_players()):
-        pm(cli, nick, messages["dullahan_targets_dead"])
+    if not TARGETS[wrapper.source] & set(get_players()):
+        wrapper.pm(messages["dullahan_targets_dead"])
         return
 
-    victim = get_victim(cli, nick, re.split(" +",rest)[0], False)
-    if not victim:
+    target = get_target(var, wrapper, re.split(" +", message)[0])
+    if not target:
         return
 
-    if victim == nick:
-        pm(cli, nick, messages["no_suicide"])
+    if target is wrapper.source:
+        wrapper.pm(messages["no_suicide"])
         return
 
-    orig = victim
-    evt = Event("targeted_command", {"target": victim, "misdirection": True, "exchange": True})
-    evt.dispatch(cli, var, "kill", nick, victim, frozenset({"detrimental"}))
+    orig = target
+    evt = Event("targeted_command", {"target": target.nick, "misdirection": True, "exchange": True})
+    evt.dispatch(wrapper.client, var, "kill", wrapper.source.nick, target.nick, frozenset({"detrimental"}))
     if evt.prevent_default:
         return
-    victim = evt.data["target"]
+    target = users._get(evt.data["target"]) # FIXME: Need to fix once targeted_command uses the new API
 
-    KILLS[nick] = victim
+    KILLS[wrapper.source] = target
 
-    msg = messages["wolf_target"].format(orig)
-    pm(cli, nick, messages["player"].format(msg))
+    wrapper.pm(messages["player"].format(messages["wolf_target"].format(orig)))
```

Comment:
This is also gross. Don't stitch together multiple messages. Make a new message for "You have selected {0} to be killed." vs "{0} has selected {1} to be killed." Both should be generic (i.e. not reference "wolf" in the key, as not only wolf kills now)

You don't need to update wolf.py as part of this PR to use the more sane messaging, but do update the above instance. You can if you want though (hunter and vigilante probably need it too).